### ISSUE TITLE
Ambiguous data in the file ndpi_content_match.c.inc

### ISSUE THREAD

**Issue by vel21ripn** [ntop/nDPI#518]
Array host_protocol_list[] have 4 ambiguous element.
```
97:    { 0xB93CDA35 /* 185.60.218.53/32 */, 27, NDPI_PROTOCOL_WHATSAPP },
219:   { 0x3452BB00 /* 52.82.0.0/14 */, 14, NDPI_PROTOCOL_AMAZON },
588:   { 0x9F7ABD30 /* 159.122.189.32 */, 21, NDPI_PROTOCOL_TEAMVIEWER },
7917:  { 0xD040C900 /* 208.64.201.0/22 */, 22, NDPI_PROTOCOL_STEAM },
```
Line 97: conflict  ip & mask
Line 219: conflict  ip & mask & comment
Line 588: conflict  ip & mask & comment
Line 7917: conflict  ip & mask

I did not check for compliance with all network addresses and comments.

Where does this list of addresses come from? Why can not it be generated from a human-readable file?

Please do not forget to update the test results after changing the list of protocols in the 'dev' branch.

**Comment by kYroL01** [comment]
test results updated, thanks to reporting.

Actually I don't understand what's the problem with that ip addresses.
That list is a about network IP addresses that belongs to various server name associated to an application.
It's handmade because i cannot see the different to put by hand the address in a file and translate in a format used in the code.
Also, I don't think is too hard to understand by human eye, isn't it ? All information you need is on the line.
Do you have better idea ?

**Comment by vel21ripn** [comment]
I'm currently porting "ndpi-netfilter" to the "dev" branch.
I got very strange errors and was forced to check the processing of the list host_protocol_list[]. Thus, the mismatch between the addresses of the networks and the mask was found, later the address and comment mismatch was found.

IMHO Comments must either match the data used, or they should not be at all.
I would like to see a one file with a list of networks by protocols and the simplest parser that would check the network and mask compliance, and also check the absence of network crossings between the protocols. I can write such a parser on the "C" for portability. The patricia library can simplify this process.

I found that libcache does not use ndpi_malloc / ndpi_calloc / ndpi_free and it also uses a mixture of declarations and code. If I make a "PR" correcting these shortcomings, will it be accepted?

There is a proposal to make an option for configuring the project, which enables / disables the output of debugging information. By default, the debug output is disabled. To enable debug information output, use the option "--enable-debug-messages". The definition of NDPI_ENABLE_DEBUG_MESSAGES changes the size of the main structure and it is very important to have this definition identical for all compiled files.
